### PR TITLE
archive build artefacts in GitHub Actions

### PR DESCRIPTION
**Pull request by dhebbeker**
Adds an action which archives the build artefacts in `build/bin/` for each build configuration. This provides the binaries for all supported operating systems for every workflow run available.

This facilitates to have development builds available for all operating systems.

Preview
----------

Screenshot from 2020-12-27 11-23-55

### CODE REVIEW

File: .github/workflows/build_cmake.yml [hunk 1]
```diff
@@ -283,6 +283,12 @@ jobs:
           message(FATAL_ERROR "Build failed")
         endif()
 
+    - name: Archive build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: "${{ matrix.config.name }} build artifacts"
+        path: build/bin/
+
```
Comment (albert-github) [inline .github/workflows/build_cmake.yml:291]: - Wouldn't this explode i.e. give 8 set of executables for each build? or will only the last be preserved?
- How can these builds be accessed?

Comment (doxygen) [reply .github/workflows/build_cmake.yml:291]: @albert-github From what I read [here](https://github.com/actions/upload-artifact) the artifacts are generated per build but are stored for a maximum of 90 days only (can be made shorter). Build artifacts are accessed via the actions page, e.g. https://github.com/doxygen/doxygen/actions/runs/446837009 under Artifacts.

Comment (doxygen) [reply .github/workflows/build_cmake.yml:291]: @dhebbeker Nice feature!

Comment (albert-github) [reply .github/workflows/build_cmake.yml:291]: Interesting article! and also the link to the artifacts of the individual build.

I see there that it says that the debug version for windows is 511 MB, looks a bit of a calculation problem (downloading it gives about 99 MB and also the release version give a factor bigger) but that is not our problem, probably the way it is stored.

I'm still a bit worried about the amount of storage that is required (looks like now per build around 1Gb) and the number of pull requests / merges done with master.

Comment (doxygen) [reply .github/workflows/build_cmake.yml:291]: @albert-github I read [here](https://docs.github.com/en/free-pro-team@latest/actions/guides/storing-workflow-data-as-artifacts):
 
> Storing artifacts uses storage space on GitHub. GitHub Actions usage is free for public repositories and self-hosted runners. 

and

> all files that were individually uploaded as part of the artifact get zipped together into a single file. This means that billing is calculated based on the size of the uploaded artifact and not the size of the zip file.

Let wait for Github to judge if we use too much data.

**Pull request merged**